pkg/gatewayserver/io/basicstationlns/config.go

```diff
@@ -0,0 +1,31 @@
+// Copyright © 2020 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package basicstationlns
+
+import "time"
+
+// Config defines the Basic Station configuration of the Gateway Server.
+type Config struct {
+	UseTrafficTLSAddress bool          `name:"use-traffic-tls-address" description:"Use WSS for the traffic address regardless of the TLS setting"`
+	WSPingInterval       time.Duration `name:"ws-ping-interval" description:"Interval to send WS ping messages"`
+	AllowUnauthenticated bool          `name:"allow-unauthenticated" description:"Allow unauthenticated connections"`
+}
+
+// DefaultConfig contains the default configuration.
+var DefaultConfig = Config{
+	UseTrafficTLSAddress: false,
+	WSPingInterval:       30 * time.Second,
+	AllowUnauthenticated: true,
```

Comment:
I don't like this. We should be secure by default.
I think security is important enough to break existing deployments that allowed unauthenticated gateways by default.

cc: @johanstokking

Reply:
I agree

Reply:
Ok but we don't do this for `gs.require-registered-gateways`. That's false by default and that's a security issue as well.

Reply:
Also I wanted this to land in a `backport` where this breaks.